### ISSUE TITLE
File description 0 not being closed

### ISSUE THREAD

**Issue by Amtrix** [jirihnidek/daemon#4]
Randomly stumbled upon this repo on how to daemonize a process. Super helpful! Found a small issue that caused me troubles. The fd = 0 is not getting closed at the line below.

https://github.com/jirihnidek/daemon/blob/d629c1fb7f395ced63fbee791ae3319c875c459f/src/daemon.c#L192

**Comment by jirihnidek** [comment]
Yeah, it is not production code. :smile: The code is intended for learning/teaching purpose. Thus I didn't check for returned values in all cases to keep code readable. Maybe I will try to fix it. BTW: Could you estimate why it wasn't possible to close `stdin`? Something like this could give you some useful error message:

```c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>
#include <errno.h>
#include <string.h>

int main(void)
{
        int ret = 0;
        printf("closing stdin\n");
        ret = close(0);
        if (ret == 0){
                printf("stdin closed\n");
        } else {
                printf("closing stdin failed: %s\n", strerror(errno));
        }
        return EXIT_SUCCESS;
}

```
